target/linux/ramips/image/mt7621.mk

```diff
@@ -420,6 +420,22 @@ define Device/re350-v1
 endef
 TARGET_DEVICES += re350-v1
 
+define Device/tplink_re650-v1
+  DTS := RE650
+  DEVICE_TITLE := TP-LINK RE650 v1
+  DEVICE_PACKAGES := kmod-mt7615e wpad-basic
+  TPLINK_BOARD_ID := RE650-V1
+  TPLINK_HWID := 0x0
+  TPLINK_HWREV := 0
+  TPLINK_HEADER_VERSION := 1
+  IMAGE_SIZE := 14208k
+  KERNEL := $(KERNEL_DTB) | tplink-v1-header -e -O
+  IMAGES += factory.bin
+  IMAGE/sysupgrade.bin := append-rootfs | tplink-safeloader sysupgrade | append-metadata | check-size $$$$(IMAGE_SIZE)
+  IMAGE/factory.bin := append-rootfs | tplink-safeloader factory
```

Comment:
Consider introducing a shared definition Device/tplink or Device/tplink-safeloader here.
Though it will be only two devices sharing them, there is a lot of duplicate lines per device.

Reply:
Okay. I'll add Device/tplink and rework re350 and re650.